### PR TITLE
Get rid of FindQt5Keychain.cmake

### PR DESCRIPTION
**Pull request by er-vin**
QtKeychain provides Qt5KeychainConfig.cmake and friends nowadays, so no
need to have a less reliable and outdated find module on our end.

Also this shows that we were including keychain.h in the wrong way and
were not using the link target, so both got fixed as well.

Signed-off-by: Kevin Ottens <kevin.ottens@nextcloud.com>

### CODE REVIEW

File: admin/linux/build-appimage.sh [hunk 1]
```diff
@@ -30,7 +30,7 @@ mkdir build
 cd build
 cmake -D CMAKE_INSTALL_PREFIX=/usr ../
 make -j4
-make DESTDIR=/app install
```
Comment (nicolasfella) [inline admin/linux/build-appimage.sh:33]: This looks unrelated?

**Review by nicolasfella — APPROVED** [review]
Looks good to me

File: admin/linux/build-appimage.sh (hunk 1 above)
Comment (er-vin) [reply admin/linux/build-appimage.sh:33]: > This looks unrelated?

Well actually not, /app wasn't in the paths the cmake was looking at to find qtkeychain.

**Review by camilasan — APPROVED** [review]
Eventually we also should update the scripts in the client-building repo, brander and mac os.

**Comment by er-vin** [comment]
> Eventually we also should update the scripts in the client-building repo, brander and mac os.

Exactly, now that it's in we should trigger all on master and see what fails.
